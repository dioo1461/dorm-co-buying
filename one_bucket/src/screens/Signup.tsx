import { stackNavigation } from '@/screens/navigation/NativeStackNavigation'
import { useNavigation } from '@react-navigation/native'
import React, { useState } from 'react'
import { View, TextInput, Button } from 'react-native'

const Signup = () => {
    const [id, setId] = useState('')
    const [phoneNumber, setPhoneNumber] = useState('')
    const [schoolName, setSchoolName] = useState('')
    const [email, setEmail] = useState('')
    const [password, setPassword] = useState('')
    const [verificationCode, setVerificationCode] = useState('')
    const navigation = stackNavigation()

    const handlePersonalVerification = () => {
        // Handle personal verification logic
    }

    const handleSchoolVerification = () => {
        // Handle school verification logic
    }

    return (
        <View>
            {/* Personal Verification */}
            <TextInput placeholder='아이디' value={id} onChangeText={setId} />
            <TextInput
                placeholder='비밀번호'
                secureTextEntry
                value={password}
                onChangeText={setPassword}
            />
            <TextInput
                placeholder='전화번호'
                value={phoneNumber}
                onChangeText={setPhoneNumber}
            />

            {/* School Verification */}
            <TextInput
                placeholder='학교명'
                value={schoolName}
                onChangeText={setSchoolName}
            />
            <TextInput
                placeholder='학교 Email'
                value={email}
                onChangeText={setEmail}
            />
            <Button title='인증 코드 전송' onPress={handleSchoolVerification} />
            {/* Resend and Edit Buttons */}
            {/* Verification Code Input */}
            <TextInput
                placeholder='인증 코드'
                value={verificationCode}
                onChangeText={setVerificationCode}
            />
            <Button
                title='다음'
                onPress={() =>
                    navigation.navigate('SetProfile', {
                        id: id,
                        password: password,
                        phoneNumber: phoneNumber,
                        schoolName: schoolName,
                        email: email,
                    })
                }
            />
        </View>
    )
}

export default Signup
