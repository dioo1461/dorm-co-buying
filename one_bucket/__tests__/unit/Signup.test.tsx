import React from 'react';
import 'react-native';
import {render, fireEvent} from '@testing-library/react-native';
import Signup from 'screens/Login';

describe('Signup', () => {
    it('renders screen Signup', () => {
        const screen = render(<Signup />);
        const json = screen.toJSON();
        expect(json).toMatchSnapshot();
    });

    it('displays title properly', () => {
        const screen = render(<Signup />);
        const title = screen.getByText('Signup');
    });
});
