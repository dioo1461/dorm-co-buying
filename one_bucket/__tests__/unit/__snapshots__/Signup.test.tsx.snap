// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup renders screen Signup 1`] = `
<View>
  <Text>
    Signup
  </Text>
</View>
`;
